src/app/posts.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+
+import { Posts } from './posts';
+import { POSTSDATA } from './posts-data';
+import { Router } from '@angular/router';
+
+@Injectable({
+	providedIn: 'root'
+})
+export class PostsService {
+	popup: Posts[];
+	filteredData: Posts[];
+	constructor(private router: Router) {}
+
+	getPosts(): Posts[] {
+		return POSTSDATA;
+	}
+	getSelectedPost(name: string): Posts[] {
+		return POSTSDATA.filter((post) => post.heading === name);
```

Comment:
if the function body following the fat arrow does not contain any curly braces then you dont need to use parenthesis